### PR TITLE
add struct and array & struct casting

### LINKED ISSUE #459: Document struct as a data type
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

SQL reference - Data types
We need to add struct as a data type, and provide necessary description and operations e.g., field access, especially because PG doesn't have it. PG only has named type `CREATE TYPE`. Struct type is like an anonymous one.

### Reference

https://www.postgresql.org/docs/current/rowtypes.html

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add docs for struct and array, and revamp the data type docs. 

- **Related code PR**: 

- **Related doc issue**: 
Resolves #459 #456 and #428 

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Comment by hengm3467** [comment]
@BugenZhao Please help review the struct casting note/description. Thanks!

**Comment by BugenZhao** [comment]
> @BugenZhao Please help review the struct casting note/description. Thanks!

@xiangjinwu Could you please help on this? Thanks. 🥰

File: docs/sql/data-types/data-type-struct.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+The statement below creates a table `x` that contains struct `a`, which contains two nested columns (`b` and `c`) that are both integers.
+
+```sql
+CREATE TABLE x (a STRUCT <b INTEGER, c INTEGER>, d INTEGER);
+```
+
+The statement below creates a table `y` that contains struct `a`, which contains another struct `c`.
+
+```sql
+CRESTE TABLE y (a STRUCT <b STRUCT<c INTEGER>, d INTEGER>, e VARCHAR);
```
Comment (xiangjinwu) [inline docs/sql/data-types/data-type-struct.md]:
```suggestion
CREATE TABLE y (a STRUCT <b STRUCT<c INTEGER>, d INTEGER>, e VARCHAR);
```

File: docs/sql/data-types/data-type-struct.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+Belos is a real world example.
```
Comment (xiangjinwu) [inline docs/sql/data-types/data-type-struct.md]:
```suggestion
Below is a real world example.
```

File: docs/sql/data-types/data-type-struct.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+### Examples
+
+```sql
+SELECT (a).b, d
+FROM x;
+```
+
+```sql
+SELECT ((a).b).c, e
+FROM y
```
Comment (xiangjinwu) [inline docs/sql/data-types/data-type-struct.md]:
```suggestion
FROM y;
```

**Review by xiangjinwu — APPROVED** [review]

File: docs/sql/data-types/data-type-array.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            'N5432N', 
+            'FAST TAXI', 
+            '2030-12-31', 
+            'DAVID WANG'
+        );
+```
+
+### Retrieve data in an array
+
+To retrieve data in an array, use the `ARRAY_COLUMN[RELATIVE_POSITION]` syntax. Relative positions start from 1. For example, to access `ABCD1234`, the first object in the `trip_id` array, we can specify trip_id[1].
```
Comment (xxchan) [inline docs/sql/data-types/data-type-array.md]:
```suggestion
To retrieve data in an array, use the `ARRAY_COLUMN[RELATIVE_POSITION]` syntax. Relative positions start from 1. For example, to access `ABCD1234`, the first object in the `trip_id` array, we can specify `trip_id[1]`.
```

File: docs/sql/data-types/data-type-array.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+#### Examples
+
+Retrieve the second element in array `a` from the `x` table.
+```sql
+SELECT a[2] FROM x;
+-----Result
+{2,3}
+```
+
+Retrieve the first array in array `trip_id` from the `taxi` table.
```
Comment (xxchan) [inline docs/sql/data-types/data-type-array.md]:
```suggestion
Retrieve the first element in array `trip_id` from the `taxi` table.
```

**Review by xxchan — APPROVED** [review]
Generally LGTM

**Pull request merged**